### ISSUE TITLE
An error/warning about NSKeyedUnarchiver

### ISSUE THREAD

**Issue by KyleXie** [aws-amplify/aws-sdk-ios#4212]
**Describe the bug**
Every time when I switch my app to background mode, I see a lot of errors/warnings from the console like this:

```
 *** -[NSKeyedUnarchiver validateAllowedClass:forKey:] allowed unarchiving safe plist type ''NSString' (0x1dedef9a8) [/System/Library/Frameworks/Foundation.framework]' for key 'NS.keys', even though it was not explicitly included in the client allowed classes set: '{(
    "'NSMutableString' (0x1dedeeb70) [/System/Library/Frameworks/Foundation.framework]",
    "'NSDictionary' (0x1dede4d10) [/System/Library/Frameworks/CoreFoundation.framework]"
)}'. This will be disallowed in the future.
```

**Expected Behavior**
No errors happen when switching app to background mode.

**Stack Trace**
I added a breakpoint, it points to this place:
![image](https://user-images.githubusercontent.com/2870241/178214303-ae75a36f-7d22-4b52-9d59-683f24ead554.png)

**Areas of the SDK you are using (AWSMobileClient, Cognito, Pinpoint, IoT, etc)?**
Pinpoint

**Environment(please complete the following information):**
 - SDK Version: 2.27.11
 - Dependency Manager: Swift Package
 - Swift Version : 5.5
 - Xcode Version: 13.4

**Device Information (please complete the following information):**
 - Device: iPhone 12
 - iOS Version: iOS 15.5

**Comment by atierian** [comment]
Thanks for opening this issue @KyleXie. We'll look into it and follow up here with any updates.

**Comment by yuryks** [comment]
Up

**Comment by ruisebas** [comment]
This has been fixed in [2.28.1](https://github.com/aws-amplify/aws-sdk-ios/releases/tag/2.28.1).
